src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -554,7 +558,11 @@ private class StatusBarProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
-                await jtf.SwitchToMainThreadAsync();
+                token.ThrowIfCancellationRequested();
```

Comment:
This seems really excessive to me. 
https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.joinabletaskfactory.switchtomainthreadasync?view=visualstudiosdk-2019 says that the check will happen in the switch to main thread.

Reply:
Check it before and pass it to the switch to main thread async?

I get the theoretical benefit, but at this point, it just adds clutter imo.

Reply:
Removed extra cancelation token call.